keyword of 'assert' in cuda

Hi~
Here is my code snippet:
```python
@cuda.jit
def kernel_fun():
      a = 1
      assert a > 1, 'error information'
```
I found the 'error information' will not be printed. Is there something wrong with me?

I think this is a documentation bug, and that `assert` in the CUDA target only works when `debug=True`. The following:

```python
from numba import cuda

@cuda.jit(debug=True)
def kernel_fun(a):
      assert a > 1, 'error information'

kernel_fun[1, 1](1)
```

produces:

```
$ python repro.py 
Traceback (most recent call last):
  File "repro.py", line 7, in <module>
    kernel_fun[1, 1](1)
  File "/home/gmarkall/numbadev/numba/numba/cuda/compiler.py", line 759, in __call__
    cfg(*args)
  File "/home/gmarkall/numbadev/numba/numba/cuda/compiler.py", line 523, in __call__
    sharedmem=self.sharedmem)
  File "/home/gmarkall/numbadev/numba/numba/cuda/compiler.py", line 630, in _kernel_call
    raise exccls(*exc_args)
AssertionError: In function 'kernel_fun', file /home/gmarkall/numbadev/issues/5196/repro.py, line 5, tid=[0, 0, 0] ctaid=[0, 0, 0]: error information
```

I think this is the right behaviour - in non-debug, I would want `assert` statements to be elided so that the overhead of them is eliminated (I suspect that checking for assertions would also mess up async launches by forcing synchronization, but I haven't yet checked). This semantic has parity with CUDA C/C++, but deviates from nopython mode on the CPU and Python in general.

I'll make a docs PR shortly that explains this.

--- a/docs/source/cuda/cudapysupported.rst
+++ b/docs/source/cuda/cudapysupported.rst
@@ -33,8 +33,13 @@ The following Python constructs are not supported:
 * Comprehensions (either list, dict, set or generator comprehensions)
 * Generator (any ``yield`` statements)
 
-The ``raise`` and ``assert`` statements are supported.
-See :ref:`nopython language support <pysupported-language>`.
+The ``raise`` statement is supported.
+
+The ``assert`` statement is supported, but only has an effect when
+``debug=True`` is passed to the :func:`numba.cuda.jit` decorator. This is
+similar to the behavior of the ``assert`` keyword in CUDA C/C++, which is
+ignored unless compiling with device debug turned on.
+
 
 Built-in types
 ===============

